lib/trento_web/controllers/catalog_controller.ex

```diff
@@ -3,6 +3,35 @@ defmodule TrentoWeb.CatalogController do
 
   alias Trento.Integration.Checks
 
+  alias TrentoWeb.OpenApi.Schema.{ChecksCatalog, Provider}
+
+  use OpenApiSpex.ControllerSpecs
+
+  tags(["Checks"])
+
+  operation(:checks_catalog,
+    summary: "Checks Catalog",
+    description:
+      "The list of the available checks that can be configured to run on the target SAP infrastructure",
+    parameters: [
+      flat: [
+        in: :query,
+        type: :string,
+        description:
+          "Whether to output a flat catalog or not. Just provide the flag, not the value. eg /api/checks/catalog?flat"
+      ],
+      provider: [
+        in: :query,
+        type: Provider.FilterableProviders,
+        description: "Whether to filter by a specific provider"
+      ]
+    ],
+    responses: [
+      ok: {"A Collection of the available Checks", "application/json", ChecksCatalog.Catalog},
```

Comment:
Are we deliveratetely not putting the `bad_request` here?
For example, we have the `bad_request` with a `not_ready` content for the moment when the catalog is being built.
This would eventually be `%{error: :not_ready}`

Reply:
Yes, my brain deliberately forgot about that :sweat_smile: 
Thanks!

Reply:
There it is dbfaff12a4f4844640730a0fa3198442dd5eec09

![image](https://user-images.githubusercontent.com/8167114/168752377-06f296a2-840a-47c5-ba33-e2b781c44d19.png)